eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,11 +319,14 @@ if ($TenantId) {
 if ($TestApplicationSecret) {
     $templateParameters.Add('testApplicationSecret', $TestApplicationSecret)
 }
+if ($ArmTemplateParameters) {
+    $templateParameters += $ArmTemplateParameters
+}
 if ($AdditionalParameters) {
     $templateParameters += $AdditionalParameters
```

Comment:
Not that we're merging two unknowns, this will fail if there's duplicates. You might want to loop over `AdditionalParameters` and overwrite instead:
```powershell
foreach ($key in $AdditionalParameters.Keys) { # No need to check for null; properties in PowerShell will never throw.
  $templateParameters[$key] = $AdditionalParameters[$Key]
}
```

Reply:
Ah, good point. In my head I didn't register the operator behavior for hashes.